packages/analytics/src/functions.ts

```diff
@@ -137,6 +138,40 @@ export async function setUserProperties(
   }
 }
 
+/**
+ * Retrieves a unique Google Analytics identifier for the web client.
+ * See {@link https://developers.google.com/analytics/devguides/collection/ga4/reference/config#client_id | client_id}.
+ *
+ * @public
+ *
+ * @param gtagFunction Wrapped gtag function that waits for fid to be set before sending an event
+ */
+export async function internalGetGoogleAnalyticsClientId(
+  gtagFunction: Gtag,
+  initializationPromise: Promise<string>
+): Promise<string> {
+  const measurementId = await initializationPromise;
+  let clientId = '';
+  if (!measurementId) {
```

Comment:
Do you need this? None of the other functions have it. There's already error handling to catch errors fetching measurementId https://github.com/firebase/firebase-js-sdk/blob/6f05b1ab5d06c2edb668de4a4d496b5d39f14329/packages/analytics/src/initialize-analytics.ts#L103